src/main/java/com/teenthofabud/wizard/nandifoods/wms/settings/payment/terms/service/impl/PaymentTermServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+    @Autowired
+    private PaymentTermFormToEntityConverter paymentTermFormToEntityConverter;
+    @Autowired
+    private PaymentTermPageDtoToPageableConverter paymentTermPageDtoToPageableConverter;
+    @Value("#{'${wms.settings.paymentTerms.search.fields}'.split(',')}") List<String> searchFields;
+
+    @Transactional
+    @Override
+    public PaymentTermVo createNewPaymentTerm(PaymentTermForm form) {
+        Optional<PaymentTermEntity> optionalPaymentTermsEntity = paymentTermRepository.findByCode(form.getCode());
+        if(optionalPaymentTermsEntity.isPresent()) {
+            throw new IllegalStateException("Payment Term already exists with id: " + form.getCode());
+        }
+
+        PaymentTermEntity paymentTermsEntity = paymentTermFormToEntityConverter.convert(form);
+
+        paymentTermsEntity = paymentTermRepository.save(paymentTermsEntity);
+
+        log.info("Payment term saved with id: {}", paymentTermsEntity.getId());
+        PaymentTermVo paymentTermVo = paymentTermEntityToVoConverter.convert(paymentTermsEntity);
+
+        return paymentTermVo;
+    }
+
+    @Override
+    public PaymentTermVo retrieveExistingPaymentTermByCode(String code) {
+        Optional<PaymentTermEntity> optionalPaymentTermsEntity = paymentTermRepository.findByCode(code);
+        if(!optionalPaymentTermsEntity.isPresent()) {
+            throw new IllegalStateException("Payment Term doesn't exist with code : " + code);
+        }
+        PaymentTermEntity paymentTermEntity = optionalPaymentTermsEntity.get();
+        log.debug("Payment term found with code: {}", paymentTermEntity.getCode());
+        PaymentTermVo paymentTermVo = paymentTermEntityToVoConverter.convert(paymentTermEntity);
+        return paymentTermVo;
+    }
+
+    @Transactional
+    @Override
+    public void deletePaymentTermByCode(String code) {
+        Optional<PaymentTermEntity> optionalPaymentTermsEntity = paymentTermRepository.findByCode(code);
+        if(!optionalPaymentTermsEntity.isPresent()) {
+            throw new IllegalStateException("Payment Term doesn't exist with code : " + code);
+        }
+        PaymentTermEntity paymentTermEntity = optionalPaymentTermsEntity.get();
+        log.debug("Payment term found with code: {}", paymentTermEntity.getCode());
+        paymentTermRepository.delete(paymentTermEntity);
+        log.info("Payment term deleted with code: {}", paymentTermEntity.getCode());
+
+    }
+
+    @Transactional
+    @Override
+    public void updatePaymentTermByCode(String code, PaymentTermDto paymentTermDto) {
+        Optional<PaymentTermEntity> optionalPaymentTermsEntity = paymentTermRepository.findByCode(code);
+        if(!optionalPaymentTermsEntity.isPresent()) {
+            throw new IllegalStateException("Payment Term doesn't exist with code : " + code);
+        }
+        PaymentTermEntity paymentTermEntity = optionalPaymentTermsEntity.get();
+        paymentTermEntity.setDaysUntilDue(paymentTermDto.getDaysUntilDue());
```

Comment:
why only one field is updated?

how would you know that the user actually intended to update something?